Saturation/SaturationAlgorithm.cpp

```diff
@@ -1189,44 +1208,52 @@ void SaturationAlgorithm::activate(Clause* cl)
   if (!cl->numSelected()) {
     TimeCounter tc(TC_LITERAL_SELECTION);
 
+    if (env.options->randomTraversals()) {
+      TimeCounter tc(TC_SHUFFLING);
+
+      Shuffling::shuffle(cl);
```

Comment:
Doesn't this mean that a clause could be shuffled a few times? At least when new, then when activated? Possibly many times with AVATAR. This might be desirable, just checking...

Reply:
Yes, the current code is definitely slightly over-shuffled. For instance, also during preprocessing we shuffle more than once (but that could have subtle implications for skolem names, etc...)